src/neo4j_graphrag/embeddings/mistral.py

```diff
@@ -0,0 +1,72 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from __future__ import annotations
+
+import os
+from typing import Any
+
+from neo4j_graphrag.embeddings.base import Embedder
+
+try:
+    from mistralai import Mistral
+except ImportError:
+    Mistral = None  # type: ignore
+
+
+class MistralAIEmbeddings(Embedder):
+    """
+    Mistral AI embeddings class.
+    This class uses the Mistral AI Python client to generate vector embeddings for text data.
+
+    Args:
+        model (str): The name of the Mistral AI text embedding model to use. Defaults to "mistral-embed".
+    """
+
+    def __init__(self, model: str = "mistral-embed", **kwargs: Any) -> None:
+        if Mistral is None:
+            raise ImportError(
+                "Could not import mistralai. "
+                "Please install it with `pip install mistralai`."
+            )
+        api_key = os.getenv("MISTRAL_API_KEY", "")
```

Comment:
We should probably allow passing the API key through kwargs as well (like for the other implementations), no? So only defaulting on getting the env var if the kwargs is not there.

Reply:
https://github.com/neo4j/neo4j-graphrag-python/pull/142/commits/8a9a307c8669038d78efb8c589d6c2d39968ef07